src/pages/index.js

```diff
@@ -66,59 +91,202 @@ function handleImageClick(card) {
 }
 
 function handleAddFormSubmit(data) {
-  renderCard(data);
-  addPicPopup.close();
+  addPicPopup.renderLoading(true);
+  api
+    .addCard(data)
+    .then((data) => {
+      renderCard(data);
+      addPicPopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      addPicPopup.renderLoading(false);
+    });
 }
 
 function handleEditProfileFormSubmit(data) {
-  userInfo.setUserInfo(data);
-  // profileName.textContent = profileModalName.value;
-  // profileTitle.textContent = profileModalTitle.value;
+  editProfilePopup.renderLoading(true);
+
+  api
+    .editUserInfo(data)
+    .then(() => {
+      userInfo.setUserInfo(data);
+      editProfilePopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      editProfilePopup.renderLoading(false);
+    });
+}
+
+let cardSection;
+
+Promise.all([api.getUserInfo(), api.getInitialCards()])
+  .then(([data, initialCards]) => {
+    console.log(data);
+    userInfo.setUserInfo(data);
+    profilePicture.src = data.avatar;
+    cardSection = new Section(
+      {
+        items: initialCards,
+        renderer: renderCard,
+      },
+      ".gallery__cards"
+    );
+    cardSection.renderItems();
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
+function handleAvatarFormSubmit(data) {
+  updateAvatar.renderLoading(true);
   console.log(data);
-  editProfilePopup.close();
+  api
+    .updateProfilePic(data)
+    .then(() => {
+      console.log(data);
+      profilePicture.src = data.link;
+      updateAvatar.close();
+    })
+    .catch((err) => console.error(err))
+    .finally(() => {
+      updateAvatar.renderLoading(false);
+    });
+}
+
+function handleAddLike(cardID) {
+  api
+    .addLike(cardID)
+    .then((data) => {
+      console.log(data);
+      this.setLikeState(data.isLiked);
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+}
+
+function handleRemoveLike(cardID) {
+  api
+    .removeLike(cardID)
+    .then((data) => {
+      console.log(data);
+      this.setLikeState(data.isLiked);
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+}
+
+function handleAttemptDelete(cardElement, cardId) {
+  console.log(cardId);
+  confirmDelete.open();
+  confirmDelete.getCardInfo(cardElement, cardId);
+}
+
+// function handleConfirmDelete(cardId) {
+//   confirmDelete.renderLoading(true);
+//   api
+//     .deleteCard(cardId)
+//     .then(() => {
+//       confirmDelete.close();
+//     })
+//     .catch((err) => console.error(err))
+//     .finally(() => {
+//       confirmDelete.renderLoading(false);
+//     });
+// }
+
+function deleteHandler(cardElement, cardId) {
+  this.renderLoading(true);
+  api
+    .deleteCard(cardId)
+    .then(() => {
+      this.close();
+    })
+    .then(() => {
+      cardElement.remove();
+      cardElement = null;
+    })
+    .catch((err) => console.error(err))
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors